mantidproject/mantid
# Disable marker colour for styles that do not use them

**Description of work.**

This commit disables the edge and face colours in the marker widget if the selected marker style does not feature those colours. For example the plus sign only uses the edge colour so the face colour is disabled

**To test:**

1. In workbench load some data and plot it.
2. Open the setitngs and go to the curves->marker tab
3. By default markers are set to 'none' when set to `none` both colour options should be disabled.
3. Change the marker to triangles, both the colours should be enabled now.
3. Change the marker to plus marks, the face colour should be disabled.
4. Check that the colours enable and disable propperly when going between markers

Fixes #26746

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
